lib/trento/application/process_managers/deregistration_process_manager.ex

```diff
@@ -84,12 +138,147 @@ defmodule Trento.DeregistrationProcessManager do
     ]
   end
 
+  def handle(
+        %DeregistrationProcessManager{
+          cluster_id: cluster_id,
+          database_instances: database_instances,
+          application_instances: application_instances
+        },
+        %HostDeregistrationRequested{
+          host_id: host_id,
+          requested_at: requested_at
+        }
+      )
+      when is_list(database_instances) and is_list(application_instances) do
+    List.flatten([
+      %DeregisterClusterHost{
+        host_id: host_id,
+        cluster_id: cluster_id,
+        deregistered_at: requested_at
+      },
+      Enum.map(database_instances, fn %Instance{
+                                        sap_system_id: sap_system_id,
+                                        instance_number: instance_number
+                                      } ->
+        %DeregisterDatabaseInstance{
+          sap_system_id: sap_system_id,
+          instance_number: instance_number,
+          host_id: host_id,
+          deregistered_at: requested_at
+        }
+      end),
+      Enum.map(application_instances, fn %Instance{
+                                           sap_system_id: sap_system_id,
+                                           instance_number: instance_number
+                                         } ->
+        %DeregisterApplicationInstance{
+          sap_system_id: sap_system_id,
+          instance_number: instance_number,
+          host_id: host_id,
+          deregistered_at: requested_at
+        }
+      end),
+      %DeregisterHost{host_id: host_id, deregistered_at: requested_at}
+    ])
+  end
+
+  def apply(
```

Comment:
I got a bit lost with the order of the applies.
We should still continue having the same order like: host - cluster - sap system function. Or grouped some other way, but I found that now they are just completely unsorted